core/chain-configs/Cargo.toml

```diff
@@ -28,7 +28,7 @@ near-time.workspace = true
 near-crypto.workspace = true
 near-o11y = { workspace = true, optional = true }
 near-parameters.workspace = true
-near-primitives.workspace = true
+near-primitives = { workspace = true, features = ["test_utils"] }
```

Comment:
Sounds super weird to enable "test_utils" for a *regular* dependency.

Reply:
That's why I initially added `sample_shard_layout` feature instead of `test_utils`.
@nagisa how strong is your opinion? @wacban wdyt?

Reply:
Yeah good point by nagisa. I can see two options:
* Replace any usages of the methods in question that are outside of tests with other logic. 
* Remove the trouble-causing randomness from those methods.